src/views/Home/components/Carousel/index.js

```diff
@@ -1,76 +1,142 @@
+  const [imageGallery, setImageGallery] = useState(null)
 
-    this.loadCarousel = this.loadCarousel.bind(this);
-
-    this.state = {
-      error: null,
-      loading: true,
-      carouselContent: {},
+  useEffect(() => {
+    const loadCarousel = async () => {
+      setLoading(true);
+      try {
+        setCarouselContent(await cms.getSlides());
+        setLoading(false);
+      } catch (error) {
+        setError(error);
+      }
     };
+    loadCarousel();
+  }, []);
 
-    this.linkArray = [];
-  }
-  componentDidMount() {
-    this.loadCarousel();
-  }
-  async loadCarousel() {
-    this.setState({ loading: true });
-    try {
-      const carouselContent = await cms.getSlides();
-      this.setState({ loading: false, carouselContent });
-      this.state.carouselContent.map((slide) => this.linkArray.push(slide.ActionLink));
-    } catch (error) {
-      this.setState({ error });
-    }
-  }
+  const handleClickSlide = useCallback(
+    () => {
+      if (carouselContent[imageGallery.getCurrentIndex()].ActionLink !== '') {
+        window.location = carouselContent[imageGallery.getCurrentIndex()].ActionLink;
+      }
+    },
+    [carouselContent, imageGallery],
+  );
 
-  handleClickSlide() {
-    if (this.state.carouselContent[this._imageGallery.getCurrentIndex()].ActionLink !== '') {
-      window.location = this.state.carouselContent[this._imageGallery.getCurrentIndex()].ActionLink;
-    }
+  if (error) {
+    throw error;
   }
 
-  render() {
-    if (this.state.error) {
-      throw this.state.error;
-    }
-
-    let content;
-    if (this.state.loading === true) {
-      content = <GordonLoader />;
-    } else {
-      const images = [];
-      this.state.carouselContent.map((slide) =>
-        images.push({
-          original: slide.ImagePath,
-          originalAlt: slide.AltTag,
-          originalTitle: slide.Title,
-        }),
-      );
-
-      content = (
-        <ImageGallery
-          ref={(i) => (this._imageGallery = i)}
-          showThumbnails={false}
-          showFullscreenButton={false}
-          showPlayButton={false}
-          showBullets={true}
-          autoPlay={true}
-          showNav={false}
-          slideInterval={5000}
-          items={images}
-          onClick={this.handleClickSlide.bind(this)}
-        />
-      );
-    }
-    return <div>{content}</div>;
+  let content;
+  if (loading === true) {
+    content = <GordonLoader />;
+  } else {
+    const images = [];
+    carouselContent.map((slide) =>
+      images.push({
+        original: slide.ImagePath,
+        originalAlt: slide.AltTag,
+        originalTitle: slide.Title,
+      }),
+    );
+
+    content = (
+      <ImageGallery
+        ref={(i) => { setImageGallery(i); }}
+        showThumbnails={false}
+        showFullscreenButton={false}
+        showPlayButton={false}
+        showBullets={true}
+        autoPlay={true}
+        showNav={false}
+        slideInterval={5000}
+        items={images}
```

Comment:
Since `images` only gets used once, you can increase the performance slightly by passing the result of `map` directly rather than storing it in an intermediate variable:
```suggestion
    content = (
      <ImageGallery
        ref={(i) => { setImageGallery(i); }}
        showThumbnails={false}
        showFullscreenButton={false}
        showPlayButton={false}
        showBullets={true}
        autoPlay={true}
        showNav={false}
        slideInterval={5000}
        items={carouselContent.map((slide) =>
             ({
                original: slide.ImagePath,
                originalAlt: slide.AltTag,
                originalTitle: slide.Title,
              })
          );
        }
```